examples/alarm/alarm.py

```diff
@@ -0,0 +1,83 @@
+import uuid
+from datetime import datetime
+
+from nisystemlink.clients.alarm import AlarmClient
+from nisystemlink.clients.alarm.models import (
+    AlarmOrderBy,
+    AlarmTransitionType,
+    CreateAlarmTransition,
+    CreateOrUpdateAlarmRequest,
+    QueryAlarmsWithFilterRequest,
+    TransitionInclusionOption,
+)
+from nisystemlink.clients.core import HttpConfiguration
+
+# Setup the server configuration to point to your instance of SystemLink Enterprise
+server_configuration = HttpConfiguration(
+    server_uri="https://yourserver.yourcompany.com",
+    api_key="YourAPIKeyGeneratedFromSystemLink",
+)
+client = AlarmClient(configuration=server_configuration)
+
+# Create a unique alarm ID for this example
+# alarm_id is a user-defined identifier for this alarm
+alarm_id = f"example_alarm_{uuid.uuid1().hex}"
+
+# Create an alarm with a SET transition
+create_request = CreateOrUpdateAlarmRequest(
+    alarm_id=alarm_id,
+    transition=CreateAlarmTransition(
+        transition_type=AlarmTransitionType.SET,
+        occurred_at=datetime.now(),
+        severity_level=3,
+        condition="Temperature exceeded threshold",
+        message="Temperature sensor reading: 85°C",
+    ),
+)
+# Returns instance_id - a server-generated unique identifier for this specific alarm occurrence
+id = client.create_or_update_alarm(create_request)
```

Comment:
A fundamental part of the alarm HTTP API is handling 409 conflicts on create or update. It enables applications/engines that produce alarms to remain stateless. We should show how to handle a 409 with comments explaining its necessity. Alternatively, we could build something into the client to handle 409s, such as returning None, but that might need to be opt-in based on a parameter.